rasterio/rio/windows.py

```diff
@@ -0,0 +1,171 @@
+"""$ rio windows"""
+
+
+import json
+import logging
+import os.path
+
+import click
+import cligj
+
+import rasterio
+from rasterio.rio import options
+from rasterio.rio.helpers import write_features
+from rasterio.warp import transform_bounds
+
+
+logger = logging.getLogger('rio')
+
+
+class _Collection(object):
+
+    """For use with `rasterio.rio.helpers.write_features()`."""
+
+    def __init__(self, src, bidx, precision, geographic):
+
+        """Produce a series of GeoJSON polygon features from a `RasterReader()`
+        representing dataset windows.
+
+        Parameters
+        ----------
+        src : RasterReader
+            An open datasource.
+        bidx : int
+            Extract windows from this band.
+        precision : int
+            Coordinate precision.
+        geographic : bool
+            Reproject geometries to ``EPSG:4326`` if ``True``.
+
+        Yields
+        ------
+        dict
+            GeoJSON polygon feature.
+        """
+
+        self._src = src
+        self._bidx = bidx
+        self._precision = precision
+        self._geographic = geographic
+
+    @property
+    def bbox(self):
+        bounds = self._src.bounds
+        if self._geographic:
+            bounds = transform_bounds(self._src.crs, 'EPSG:4326', *bounds)
+        if self._precision > 0:
```

Comment:
Never mind - not broken in `$ rio shapes`.  This option is defined in `cligj` and has a default of `-1` (not `None`), so `if precision >= 0` fixed it.